### PR TITLE
RAINCATCH-1190 - Extract security interface

### PR DESCRIPTION
**Pull request by wtrocki**

### CODE REVIEW

File: demo/server/src/modules/keycloak/index.ts [hunk 1]
```diff
@@ -1,26 +1,38 @@
...
-  // Express Session Configuration.
-  app.use(session(sessionOpts));
-
-  // Create a session store
-  const memoryStore = new session.MemoryStore();
-  const keycloak = new Keycloak({ store: memoryStore }, config.keycloakConfig);
-
-  // Use keycloak middleware & define applications logout route
-  app.use(keycloak.middleware({logout: '/logout'}));
+export class KeycloakSecurity implements EndpointSecurity {
```
Comment (wtrocki) [inline demo/server/src/modules/keycloak/index.ts:13]: I still think that despite this being really simple we should have it as separate module.

File: demo/server/src/modules/keycloak/index.ts [hunk 2]
```diff
@@ -1,26 +1,38 @@
...
+  private keycloak: any;
+
+  public init(app: express.Express, sessionOpts: session.SessionOptions): void {
+    // Express Session Configuration.
+    app.use(session(sessionOpts));
+
+    // Create a session store
+    const memoryStore = new session.MemoryStore();
+    this.keycloak = new Keycloak({ store: memoryStore }, config.keycloakConfig);
+  }
```
Comment (JameelB) [inline demo/server/src/modules/keycloak/index.ts]: also re-add the `app.use(keycloak.middleware({logout: '/logout'}));` to use keycloak middleware and define the logout route

File: cloud/auth/src/EndpointSecurity.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  protect(role?: string): express.Handler;
+
+  /**
+   * Create middleware for authentication purposes
+   * This method wraps `passport.authenticate` to provide middleware for authenticating users.
+   *
+   * @param redirect - location to redirect after successful authentication
+   * @param loginError - location to redirect after failed authentication
+   */
+  authenticate(redirect: string, loginError: string): express.Handler;
```
Comment (JameelB) [inline cloud/auth/src/EndpointSecurity.ts]: would it be possible to remove this from the interface if this only refers to passport?

Comment (wtrocki) [inline cloud/auth/src/EndpointSecurity.ts]: Everything for you ;P

**Comment by wtrocki** [comment]
@JameelB Really good point about dropping this method. 
It looks like we using that in top level application anyway.

**Review by paolobueno — APPROVED** [review]

**Pull request merged**